library/Mockery/Reflector.php

```diff
@@ -249,6 +251,11 @@ private static function getTypeFromReflectionType(\ReflectionType $type, \Reflec
                 $type->getTypes()
             );
 
+            $intersect = array_intersect(self::TRAVERSABLE_OBJECT_ARRAY, $types);
+            if (self::TRAVERSABLE_OBJECT_ARRAY === $intersect) {
```

Comment:
I'm trying to determine a more appropriate question to ask in this context. 

Instead of asking if the list of `$types` contains all the values of `TRAVERSABLE_OBJECT_ARRAY`, and if so, replacing it with the value of `ITERABLE_OBJECT` and merging any other values from `$types`

should we ask if the list of `$types` contains the value `object` and any type that is prefixed with `\`, indicating it is a class(object)?

Reply:
> I think the "easiest" way to fix that one, is just to keep track if both `\Traversable` and `array` are in the type list and just drop those and convert them to `iterable`.
>
> But yeah you could also ask that question, that if you encounter `object` you can/must discard other class-types
>
_Originally posted by @girgias in [discord](https://discord.com/channels/356354025865740288/862798982936789013/1134848809885765683)._